src/IR/LambdaLift.hs

```diff
@@ -85,17 +93,26 @@ addCurrentScope :: I.VarId -> LiftFn ()
 addCurrentScope v =
   modify $ \st -> st { currentScope = S.insert v $ currentScope st }
 
--- | Get a fresh variable name
-getFresh :: LiftFn String
+-- | Create a string composed of the scope trail and a variable name.
+prependTrail :: Identifier -> LiftFn Identifier
+prependTrail cur = do
+  curTrail <- gets trail
+  return $ foldr (<>)
+                 (Identifier "")
+                 (intersperse (Identifier "_") (reverse (cur : curTrail)))
+
+-- | Construct a fresh variable name for a new lifted lambda.
+getFresh :: LiftFn Identifier
 getFresh = do
   curCount <- gets anonCount
+  let anonName = Identifier "anon" <> fromString (show curCount)
```

Comment:
Remove `Identifier`

Reply:
Also, there's no need to use a `let` binding here; just inline it at the `return`